services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/links/LinkMaintenanceSpec.groovy

```diff
@@ -0,0 +1,92 @@
+        islUtils.getIslCost(isl) == Constants.DEFAULT_COST
+        islUtils.getIslCost(islUtils.reverseIsl(isl)) == Constants.DEFAULT_COST
+    }
+
+    def "Flows can be evacuated (rerouted) when setting maintenance mode for a particular link"() {
+        given: "Two active not neighboring switches with two possible paths at least"
+        def switches = topology.getActiveSwitches()
+        def allLinks = northbound.getAllLinks()
+        List<List<PathNode>> possibleFlowPaths = []
+        def (Switch srcSwitch, Switch dstSwitch) = [switches, switches].combinations()
+                .findAll { src, dst -> src != dst }.find { Switch src, Switch dst ->
+            possibleFlowPaths = database.getPaths(src.dpId, dst.dpId)*.path.sort { it.size() }
+            allLinks.every { link ->
+                !(link.source.switchId == src.dpId && link.destination.switchId == dst.dpId)
+            } && possibleFlowPaths.size() > 1
+        } ?: assumeTrue("No suiting switches found", false)
+
+        and: "Make the first path more preferable than others"
+        possibleFlowPaths[1..-1].each { pathHelper.makePathMorePreferable(possibleFlowPaths.first(), it) }
+
+        and: "A couple of flows built through these switches"
+        def flow1 = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flowHelper.addFlow(flow1)
+        def flow1Path = PathHelper.convert(northbound.getFlowPath(flow1.id))
+
+        def flow2 = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flowHelper.addFlow(flow2)
+        def flow2Path = PathHelper.convert(northbound.getFlowPath(flow2.id))
+
+        assert flow1Path == possibleFlowPaths.first()
+        assert flow2Path == possibleFlowPaths.first()
+
+        when: "Set maintenance mode without flows evacuation flag for the first link involved in flow paths"
+        def isl = pathHelper.getInvolvedIsls(flow1Path).first()
+        northbound.updateLinkUnderMaintenance(islUtils.getLinkUnderMaintenance(isl, true, false))
+
+        then: "Flows are not evacuated (rerouted) and have the same paths"
+        PathHelper.convert(northbound.getFlowPath(flow1.id)) == flow1Path
+        PathHelper.convert(northbound.getFlowPath(flow2.id)) == flow2Path
+
+        when: "Unset maintenance mode and set again with flows evacuation flag for the same link"
+        northbound.updateLinkUnderMaintenance(islUtils.getLinkUnderMaintenance(isl, false, false))
+        northbound.updateLinkUnderMaintenance(islUtils.getLinkUnderMaintenance(isl, true, true))
+
+        then: "Flows are evacuated (rerouted)"
+        PathHelper.convert(northbound.getFlowPath(flow1.id)) != flow1Path
+        PathHelper.convert(northbound.getFlowPath(flow2.id)) != flow2Path
+
+        and: "Delete flows, unset maintenance mode and reset costs"
+        [flow1, flow2].each { flowHelper.deleteFlow(it.id) }
+        northbound.updateLinkUnderMaintenance(islUtils.getLinkUnderMaintenance(isl, false, false))
+        database.resetCosts()
```

Comment:
this should not be required I guess

Reply:
At L57 we use pathHelper.makePathMorePreferable(), so we need to cleanup costs.